wsd/RequestVettingStation.cpp

```diff
@@ -489,12 +489,17 @@ void RequestVettingStation::createClientSession(const std::shared_ptr<DocumentBr
     std::shared_ptr<std::unique_ptr<WopiStorage::WOPIFileInfo>> wopiFileInfo =
         std::make_shared<std::unique_ptr<WopiStorage::WOPIFileInfo>>(std::move(realWopiFileInfo));
 
+    // _socket is now the DocumentBroker poll's responsibility forget about it here
+    std::shared_ptr<StreamSocket> socket = _socket;
```

Comment:
I get that streamSocket::setHandler replaces ClientRequestDispatcher and that the setupTransfer can happen in its own thread causing ClientRequestDispatcher itself to be already deleted, and taking RequestVettingStation with it, when RequestVettingStation's _socket.reset() was called in the pre-patch case.